crates/common/tedge_config/src/tedge_config_cli/figment.rs

```diff
@@ -11,10 +11,57 @@ use serde::de::DeserializeOwned;
 
 use crate::TEdgeConfigError;
 
+pub trait ConfigSources {
+    fn include_enviroment(&self) -> bool;
+    fn new() -> Self
+    where
+        Self: Sized;
+}
```

Comment:
I don't understand the point of this `new` method as all the known implementations, `FileAndEnvironment` and `FileOnly` can be directly build as `FileAndEnvironment` and `FileOnly` without any function call.

Reply:
This is to allow use to create the value from a generic type implementing `ConfigSources`, although I don't think I really gain anything by using `&dyn ConfigSources` in the `extract_data` function, I could just make that generic too and this issue goes away entirely.

Reply:
Yep, I think this can be make simpler but this is not really important. => Keep this as it is now.